pkg/controller/cronjob/utils.go

```diff
@@ -148,6 +148,33 @@ func getRecentUnmetScheduleTimes(sj batchv1beta1.CronJob, now time.Time) ([]time
 	return starts, nil
 }
 
+// LatestScheduleTime, foundSchedule
+// getLatestMissedSchedule returns the latest start time in the specified time window (inclusive), and a true/false indicating is a start time was found.
+// It uses a binary search with the cron.Next function, to avoid stepping through every cron schedule.
+// This is less efficient than if cron.Previous existed, but is arguable simpler to test than a custom Previous function.
+func getLatestMissedSchedule(startWindow time.Time, endWindow time.Time, schedule cron.Schedule) (time.Time, bool) {
+	minimumTimeUnit := time.Minute // Crons can't schedule in lower granularity than a minute
+	if endWindow.Sub(startWindow) < minimumTimeUnit { // Do nothing, we can't get more granular.
+		return time.Time{}, false
+	}
+
+	midway := startWindow.Add(endWindow.Sub(startWindow)/2)
+	nextScheduleAfterMidway := schedule.Next(midway)
+
+	if !nextScheduleAfterMidway.After(endWindow) { // Has a start time in the second half.
```

Comment:
Need to account for the 0-time.

Reply:
yeah i think || IsZero is enough?

Reply:
also — I believe if the nextSchedule is _after_ the end, then it means we should look on the left side

Reply:
That's already this behavior (after the end -> hits the else of 167), but perhaps it could be structured more clearly.